pkg/catalog/types.go

```diff
@@ -0,0 +1,48 @@
+package catalog
+
+import (
+	imagev1 "github.com/openshift/api/image/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ComponentType ...
+type ComponentType struct {
```

Comment:
Why not just `Component`?

Reply:
Because component is something else. Component is created with `odo component create` and it represents the running application. 
This struc is mean to represent the type of the component (nodejs, java, php ....)